stock-service/src/main/java/rs/banka4/stock_service/runners/TestDataRunner.java

```diff
@@ -238,192 +71,471 @@ public void run(String... args) {
+        long ms_id = 0;
+        List<ForexPair> forexPairs = new ArrayList<>();
+        for (CurrencyCode currencyCode1 : CurrencyCode.values()) {
+            for (CurrencyCode currencyCode2 : CurrencyCode.values()) {
+                if (currencyCode1.equals(currencyCode2)) {
+                    continue;
+                }
+
+                String ticker = currencyCode1.name() + "/" + currencyCode2.name();
+
+                double val = -1.0;
+                try {
+                    double v1 =
+                        getCurrencyValue(
+                            currencyCode1.name()
+                                .toUpperCase(),
+                            exchangeData
+                        );
+                    double v2 =
+                        getCurrencyValue(
+                            currencyCode2.name()
+                                .toUpperCase(),
+                            exchangeData
+                        );
+                    val = v1 / v2;
+                } catch (JSONException e) {
+                    LOGGER.error("Json exception in calculating forex pair value");
+                }
+
+                forexPairs.add(
+                    ForexPair.builder()
+                        .id(new UUID(ms_id, id++))
+                        .baseCurrency(currencyCode1)
+                        .quoteCurrency(currencyCode2)
+                        .liquidity(forexLiquidities[i++])
+                        .exchangeRate(new BigDecimal(val))
+                        .ticker(ticker.toUpperCase())
+                        .name(ticker)
+                        .build()
+                );
+                i = i % 3;
+            }
+        }
+        forexPairRepository.saveAllAndFlush(forexPairs);
         LOGGER.info("Production forex pairs seeded successfully");
     }
 
+
+    private void seedForexPairs() {
+        List<String> forexPairs = new ArrayList<>();
+        for (int i = 0; i < CurrencyCode.values().length; i++) {
+            for (int j = 0; j < CurrencyCode.values().length; j++) {
+                if (i == j) continue;
+
+                // FETCHING THE FOREX PAIRS
+                String url =
+                    "https://www.alphavantage.co/query?function=CURRENCY_EXCHANGE_RATE&from_currency="
+                        + CurrencyCode.values()[i]
+                        + "&to_currency="
+                        + CurrencyCode.values()[j]
+                        + "&apikey="
+                        + vantageKey;
+                Request request =
+                    new Request.Builder().url(url)
+                        .addHeader("Content-Type", "application/json")
+                        .addHeader("Authorization", vantageKey)
+                        .build();
+                try {
+                    Response response =
+                        stockHttpClient.newCall(request)
+                            .execute();
+                    assert response.body() != null;
```

Comment:
never use `assert` for anything but assumptions - can the client return `null` as a body?  if so, this isn't correct